File: AndroidStudioProjects/Ch4PracticeSet/app/src/main/java/com/example/ch4practiceset/MainActivity.java
package com.example.ch4practiceset;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    private Button next;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button next=(Button) findViewById(R.id.main_button);
        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d("General","Button clicked");
                Intent intent = new Intent(MainActivity.this,WebsiteLinks.class);
                startActivity(intent);
            }
        });

    }
}

File: CoroutinesPractice1/app/src/main/java/com/example/coroutinespractice1/MainActivity.kt
package com.example.coroutinespractice1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        CoroutineScope(Dispatchers.Main).launch{
            Log.d("General" , "Thread ${Thread.currentThread().name}")
        }

        CoroutineScope(Dispatchers.IO).launch{
            Log.d("General" , "Thread ${Thread.currentThread().name}")
        }





    }
}